detectable/src/main/java/com/synopsys/integration/detectable/detectables/go/gomod/GoModCliExtractor.java

```diff
@@ -78,20 +80,8 @@ public Extraction extract(final File directory, final File goExe) {
 
     private List<String> modGraphOutputWithReplacements(File directory, File goExe, List<String> listUJsonOutput) throws ExecutableRunnerException, DetectableException {
         final List<String> modGraphOutput = execute(directory, goExe, "Querying for the go mod graph failed:", "mod", "graph");
-        String jsonString = convertOutputToJsonString(listUJsonOutput);
-
-        Type goListUJsonEntryType = new TypeToken<List<GoListUJsonData>>() {}.getType();
-        List<GoListUJsonData> data = gson.fromJson(jsonString, goListUJsonEntryType);
 
-        for (final GoListUJsonData entry : data) {
-            ReplaceData replace = entry.getReplace();
-            if (replace != null) {
-                String path = entry.getPath();
-                String originalVersion = entry.getVersion();
-                String replaceVersion = replace.getVersion();
-                replacementData.put(String.format("%s@%s", path, originalVersion), String.format("%s@%s", path, replaceVersion));
-            }
-        }
+        replacementDataExtractorC.extractReplacementData(listUJsonOutput);
```

Comment:
I prefer replacementDataExtractorC but he should return the replacement map (and not be passed a shared map). 
Map<String, String> replacementData = replacementDataExtractorC.extractReplacementData(listUJsonOutput);